local-cli/link/link.js

```diff
@@ -76,31 +74,40 @@ const linkDependencyAndroid = (androidProject, dependency) => {
   });
 };
 
-const linkDependencyWindows = (windowsProject, dependency) => {
-
-  if (!windowsProject || !dependency.config.windows) {
-    return null;
-  }
-
-  const isInstalled = isInstalledWindows(windowsProject, dependency.config.windows);
-
-  if (isInstalled) {
-    log.info(chalk.grey(`Windows module ${dependency.name} is already linked`));
-    return null;
-  }
-
-  return pollParams(dependency.config.params).then(params => {
-    log.info(`Linking ${dependency.name} windows dependency`);
-
-    registerDependencyWindows(
-      dependency.name,
-      dependency.config.windows,
-      params,
-      windowsProject
-    );
-
-    log.info(`Windows module ${dependency.name} has been successfully linked`);
-  });
+const linkDependencyPlatforms = (platforms, project, dependency) => {
+  const ignorePlatforms = ['android', 'ios'];
+  Object.keys(platforms || [])
```

Comment:
Okay, so the approach you have taken here is to explicitly do platform `link`. How about making it so that when plugin defines `link` command, it will be run like this:
```js
originalLink().then(linkOfPlugin)
```
That way, we could not only allow `react-native-windows` to hook into `link` but potentially `init` and others. It would be also a bit less hardcoded then the following code.

Reply:
@grabbou - for most commands there wouldn't be any problems with this.  `commander` doesn't enforce that one command to one key. E.g., I could easily define `link` in the `react-native-windows` project, and `commander` would just execute both the `link` from `react-native` and from `react-native-windows`. *However* - the semantics of `link` are as follows:
1. Run prelink scripts
2. Run link for all platforms
3. Run postlink scripts

If I did `originalLink().then(linkOfPlugin)`, we would probably need to breakdown prelink and postlink per platform as well (not the worst idea), e.g.:
1. prelink for iOS
2. link iOS
3. postlink for iOS
4. prelink for Android
...

That is a much more involved change that I didn't want to undertake.

Also - `init` is a very difficult case because it's actually implemented partially by the `react-native-cli` global tool, and we currently do some heuristics to match the version of `react-native-windows` to the current installed version of `react-native`.  Not saying its impossible, but I think `init` will require some other infrastructure. For other commands, there's no reason why we can't just define another `upgrade`, `eject`, `info` and just have it work as you propose.